### PR TITLE
feat: 新增 SMTP 邮件投递功能

### PR DESCRIPTION
**Pull request by FFFold**
## 变更概述

本次改动为插件新增了基于 SMTP 的邮件投递能力，支持将整本漫画或指定章节打包后发送到指定邮箱，并补齐了该功能所需的配置、文档、错误处理、配额控制与测试覆盖。

### 1. 新增邮件投递命令

新增两条命令：

- `/jmemail <ID> <邮箱>`：下载整本并发送到邮箱
- `/jmcemail <本子ID> <章节序号> <邮箱>`：下载指定章节并发送到邮箱

邮件命令复用了现有下载与打包逻辑，支持当前插件已有的：

- ZIP / PDF 打包格式
- 打包密码
- 文件命名规则
- 下载后的自动清理

### 2. 新增 SMTP 邮件发送模块

新增独立的邮件发送模块 `core/mailer.py`，用于处理 SMTP 附件发送流程，支持：

- SMTP 配置合法性校验
- SSL / STARTTLS 连接模式
- 附件大小限制校验
- 邮件主题与正文模板渲染
- 打包文件作为邮件附件发送

同时在插件主流程中完成接入，使邮件功能可以直接复用现有下载结果。

### 3. 新增 SMTP 配置项与配置读取能力

在配置层补充了邮件发送所需的完整配置项，包括：

- `smtp_enabled`
- `smtp_host`
- `smtp_port`
- `smtp_use_ssl`
- `smtp_use_tls`
- `smtp_username`
- `smtp_password`
- `smtp_from_email`
- `smtp_from_name`
- `email_subject_template`
- `email_body_template`
- `email_max_attachment_mb`
- `email_send_timeout`

并同步更新：

- `core/base/config.py`
- `_conf_schema.json`
- 测试配置样例

### 4. 完善邮件命令执行流程

围绕邮件命令的实际使用场景，对执行流程做了补全和修正：

- 在下载前先校验邮件发送前置条件，避免 SMTP 未配置或 `pack_format=none` 时仍继续下载
- 只有在邮件实际发送成功后才消耗每日下载配额，避免“没收到邮件却被扣额度”
- 对邮件模板格式错误提供明确报错，避免误报为下载失败
- 失败时保留本地文件，方便排查和二次处理

### 5. 保持邮件命令与原有下载命令的交互一致性

为了让新增命令的使用体验与现有命令一致，本次也对聊天侧交互做了对齐：

- `/jmemail` 的交互流程与 `/jm` 保持一致
  - 开始下载提示
  - 封面 / 漫画详情预览
  - 复用现有的预览撤回逻辑

- `/jmcemail` 的交互流程与 `/jmc` 保持一致
  - 获取章节信息提示
  - 章节标题与章节序号提示
  - 然后继续下载并发送到邮箱

### 6. 更新文档说明

更新 `README.md`，补充：

- 新命令使用方式
- SMTP 配置项说明
- 邮件发送配置示例
- 邮件发送行为说明
- 常见问题说明

使用户可以直接根据文档完成配置和使用。

### 7. 补充测试覆盖

新增和更新了与邮件功能相关的单元测试，覆盖内容包括：

- SMTP 配置读取
- 附件大小限制
- 邮件发送成功 / 失败场景
- 邮件命令前置校验
- 配额扣减时机
- 模板格式错误处理
- `/jmemail` 与 `/jm` 的交互一致性
- `/jmcemail` 与 `/jmc` 的交互一致性

## 测试

执行命令：

```bash
uv run --with pytest --with pytest-asyncio python -m pytest tests/unit -q
```

结果：

- `119 passed`

## 影响范围

本次 PR 主要是新增邮件投递能力，并补齐其完整配套支持，涉及：

- 命令入口
- 核心邮件发送模块
- 配置项与 schema
- 文档
- 单元测试

对原有下载、搜索、浏览等既有功能没有引入破坏性改动。

## Summary by Sourcery

Add SMTP-based email delivery for downloaded comics and refactor common download/pack/send flows for reuse across chat and email commands.

New Features:
- Introduce SMTP mailer module for sending downloaded comic archives as email attachments.
- Add /jmemail and /jmcemail commands to download full albums or specific chapters and send them to a specified email address.

Enhancements:
- Refactor download, packing, quota checking, and cover preview logic into reusable helpers shared by existing and new commands.
- Extend message formatting to report email send success and failure details, including recipient information.
- Ensure quotas are only consumed after successful email delivery and keep local files on send failures for troubleshooting.

Documentation:
- Update README with new email commands, SMTP configuration options, usage examples, and FAQ entries about common mail issues.
- Document the new mailer module in the project structure overview.

Tests:
- Add unit tests for SMTP configuration and attachment validation, synchronous and async email sending flows, and main email command behavior and edge cases.
- Extend existing config and formatter tests to cover new SMTP-related settings and email result formatting.
- Add AstrBot and core module mocks in test conftest to support testing the main plugin entrypoints.

### CODE REVIEW

**Comment by FFFold** [comment]
@sourcery-ai review

File: main.py [hunk 1]
```diff
@@ -93,6 +98,210 @@ def _check_permission(self, event: AstrMessageEvent) -> tuple[bool, str]:
...
+        if limit > 0 and not is_admin:
+            can_download, used, total = self.quota_manager.check_quota(user_id, limit)
+            if not can_download:
+                return (
+                    False,
+                    f"❌ 今日下载次数已达上限 ({used}/{total})\n请明天再试",
+                    limit,
+                )
+
+        return True, "", limit
```
[the remark this answers is not in the document]
Comment (FFFold) [reply main.py:145]: 不太理解，既然已经将admin排除出配额检查，那是否设置这个sentinel limit有什么意义呢

[the remark this answers is not in the document]
Comment (FFFold) [reply main.py:145]: 我检查了原有的/jm 和 /jmc两个指令的配额消费逻辑，当前邮件命令的配额消费逻辑是与现有下载命令保持一致的，因此这里不需要修改，应当由上游维护者解决这个潜在的风险点。

具体来说，`_check_download_quota()` 返回的 `limit` 会被后续命令统一用于判断是否调用 `consume_quota()`；这一模式不仅用于 `/jmemail` 和 `/jmcemail`，也同样用于原有的 `/jm` 和 `/jmc`。

相关代码位置：
- 配额检查逻辑：`main.py:106-119`
- `/jm` 配额消费：`main.py:340-365`
- `/jmc` 配额消费：`main.py:434-475`
- `/jmemail` 配额消费：`main.py:542-573`
- `/jmcemail` 配额消费：`main.py:643-690`

也就是说，管理员“跳过限额检查但仍可能被记账”的行为不是邮件命令新增的问题，而是当前四个下载命令共享的既有逻辑。为了保持行为一致性，这里先不单独调整邮件命令；如果需要修复，建议作为独立改动统一调整 `/jm`、`/jmc`、`/jmemail`、`/jmcemail`` 的配额语义。

**Comment by FFFold** [comment]
AI reviewer意见均已修复，另外请维护者注意最后一条review意见。

**Comment by FFFold** [comment]
@sourcery-ai review

**Review by SourceryAI — COMMENTED** [review]
Hey - I've reviewed your changes and they look great!

***

Hi @FFFold! 👋

Thanks for trying out Sourcery by commenting with `@sourcery-ai review`! 🚀

Install the [sourcery-ai bot](https://github.com/apps/sourcery-ai/installations/new) to get automatic code reviews on every pull request ✨

<sub>
Help me be more useful! Please click 👍 or 👎 on each comment and I'll use the feedback to improve your reviews.
</sub>
